src/controllers/authController.js

```diff
@@ -158,9 +158,47 @@ const forgotPassword = async (req, res) => {
   res.status(StatusCodes.OK).json({ message: 'Password reset email sent' });
 };
 
+//reset Password endpoint
+const reset_password = async (req, res) => {
+  const { passwordToken, newPassword } = req.body;
+  //AKOS: How would we receive this passwordToken from the frontend?
```

Comment:
When the user receives their "forgot password" email, they were given a link to the "reset password" page that included the token as a query parameter: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7195c0c10fe8f2cfeee15e887c094408524535e6/src/mailing/sender.js#L54](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7195c0c10fe8f2cfeee15e887c094408524535e6/src/mailing/sender.js#L54)

The frontend should pass that query parameter back to the backend in this `resetPassword` endpoint.

Reply:
@akosasante thank you!! just added to .env BACKEND_BASE_URL_AUTH_PROD=https://musicstore.onrender.com/api/v1/auth
BACKEND_BASE_URL_AUTH_DEV=https://localhost:8000/api/v1/auth

Reply:
@akosasante so this part is wrong- right? if frontend passes 'token' query parameter back to the backend in this resetPassword endpoint- it doesn't come from req.body, but from req.query.token?
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/aea70a73-dced-462d-89ee-702223040041) so we could modify this: 
  ```
const {  newPassword } = req.body; 
  const { passwordToken} = req.query.token ;
```
right?

Reply:
The user will be brought to a page: `<url>?token=<resetToken>`. The page will have inputs for them to type in their new password and submit the form. The frontend can choose to pass that data to this endpoint in the request body; it doesn't necessarily have to pass it to the endpoint in the query parameter just because it received the info in a query parameter. Whichever is easier. I would suggest that the `body` is more common for a POST request like this. 

Example vanilla js code:

```js
const urlParams = new URLSearchParams(window.location.search);
const token = urlParams.get('token');

await fetch("/api/v1/auth/reset_password", {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    newPassword: "newPassword",
    passwordToken: token,
  }),
});

```

Reply:
@akosasante thank you so much for the explanation!!!